### PR TITLE
module 5: hooks

### PR DESCRIPTION
**Pull request by d1rtych**

### CODE REVIEW

File: src/components/AppBody/AppBody.tsx [hunk 1]
```diff
@@ -8,26 +9,56 @@ import { ContentStyled, MoviesFilter, MoviesGrid } from '../../App.styled';
...
+  const sortBy = searchParams.get('sortBy') || INITIAL_SORTING_OPTIONS;
+  const search = searchParams.get('search') || undefined;
+  const [selectedGenre, setSelectedGenre] = useState(genre);
+  const [selectedSort, setSelectedSort] = useState(sortBy);
+  const params = useMemo(() => ({
+    filter: genre !== 'All' ? genre : undefined,
+    sortBy: sortBy,
+    search: search ? search : undefined,
+    searchBy: search ? 'title' : undefined,
+  }), [genre, sortBy, search]);
```
Comment (valentynkinakh) [inline src/components/AppBody/AppBody.tsx]: **nitpick:** I think we can extract this logic in new hook (e.g. `useMoviesFilter`)

**Review by valentynkinakh — APPROVED** [review]

File: src/shared/hooks/useMovies.tsx [hunk 2]
```diff
@@ -1,20 +1,14 @@
...
-      })
-      .catch(error => {
-        console.error(error);
-      })
-  }, []);
-
-  return { movies };
-}
+  return { movies, isLoading, error };
+};
```
Comment (valentynkinakh) [inline src/shared/hooks/useMovies.tsx]: **nitpick:** It's better to return `useQuery` from this hook

File: src/components/AppBody/AppBody.tsx (hunk 1 above)
Comment (d1rtych) [reply src/components/AppBody/AppBody.tsx]: done

File: src/shared/hooks/useMovies.tsx (hunk 2 above)
Comment (d1rtych) [reply src/shared/hooks/useMovies.tsx]: done

**Pull request merged**